docs: update readme and contributing files

mds1 opened the pull request:
Resolves #164

apbendi commented on CONTRIBUTING.md:
```diff
@@ -155,6 +162,62 @@ git push origin HEAD # This will close the PR
 * Keeps commit history clean and easy to read
 
 
+## App Contribution Guidelines
```
Wondering out loud if we should change the title of this section to "Frontend Contribution Guidelines" for extra clarity. There's some ambiguity about whether a web3 "app" means just the frontend or frontend + contracts.

mds1 replied:
Should we rename the root folder from `app` to `frontend` then? I went with app here so they'd be consistent

apbendi replied:
I think it's ok to leave the folder named `app`, but for someone not yet familiar with that naming convention, using Frontend here might be clearer. Don't feel strongly about it, happy to hear what others think and fine either way.

mds1 replied:
Updated to frontend in 42a8e96d95f2138fea0243f7438bb8c5d7f4f6a6

apbendi commented on CONTRIBUTING.md:
```diff
@@ -155,6 +162,62 @@ git push origin HEAD # This will close the PR
+
+Check out [`GrantRegistryNewGrant.vue`](app/src/views/GrantRegistryNewGrant.vue) for an example of a component that uses the Composition API.
+In the Composition API docs referenced earlier, most examples put everything within the `setup()` method.
+This is ok, but typically should only be done for simple components. For more complex components, logic and variables should be moved in composition functions, like the `useNewGrant()` method defined in `GrantRegistryNewGrant.vue`.
```
Suggested simplification for the first sentence: "This is ok for simple components."

mds1 replied:
Done in 42a8e96d95f2138fea0243f7438bb8c5d7f4f6a6

apbendi reviewed (approved):
Left some very minor comments, but looks great to me! I'll leave it to @thelostone-mc and @gdixon to weigh in the substance, since I'm not a savvy enough frontend dev to have strong opinions.

carletex reviewed (commented):
This is great @mds1! Would have saved a lot time if we had this from the beginning :)

Should we add something about Tailwind? Like when to use utilities, variants. etc. I don't have much experience with it, but maybe someone can chime in and write set some Tailwind standards.

apbendi commented:
@mds1 maybe you can include a correction to the error you found in my rebasing guide here as well?

mds1 commented:
> @mds1 maybe you can include a correction to the error you found in my rebasing guide here as well?

@apbendi Done in 42a8e96d95f2138fea0243f7438bb8c5d7f4f6a6

> Should we add something about Tailwind? Like when to use utilities, variants. etc. I don't have much experience with it, but maybe someone can chime in and set some Tailwind standards.

@carletex Great call, added info about that in d72ebbd59a987297e92057dba00adeea57d06275, and also added info about the Vue style guide in 88dac00

mk1r commented:
@mds1 Looking really good. I would just suggest to add a note somewhere in the rebase section to make it clear not to make any changes to the files when resolving conflicts, unless the changes are directly related to the rebase.

mk1r commented on CONTRIBUTING.md:
````diff
@@ -131,7 +140,7 @@ The developer or developers working on said feature are responsible for keeping
-git rebase my-feature-branch main
+git rebase main
 ```
 
````
I would just suggest to add a note somewhere in the rebase section to make it clear not to make any changes to the files when resolving conflicts, unless the changes are directly related to the rebase.

mds1 replied:
Is there a phrasing you think would be more useful than what we have here: https://github.com/dcgtc/dgrants/pull/167/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R146

It says:

> If conflicts are found, you will have to resolve them now. See here for more info on resolving merge conflicts when rebasing. After successfully rebasing, force push back to your feature branch, but never force push to main:

mk1r replied:
hmm, yeah I guess I'll have to figure out better phrasing - this seems a bit confusing to me, since it seems like it's saying to resolve merge conflicts during rebase. I think it would be better if it was explicitly clear that you can't really resolve merge conflicts (things that break other parts of the code because of new commits) until after rebasing is done. 

Even in our Discord chat, the wording is more clear:

> I think a better phrasing would be "no changes outside the merge conflict"
> But fixing things within the conflict is ok

mds1 replied:
@apbendi any thoughts here?

apbendi replied:
Added the following:

> In some circumstances, the process of rebasing and resolving merge conflicts may have broken the feature or functionality you were working on. This tends to occur when large changes were made on `main` to the same section of code being modified in a feature branch. In these cases, simply add additional commit(s) to the head of your feature branch correcting the problems.

Going to merge this PR, but if we feel there is still room for improvement, feel free to open an issue or correcting PR @mds1 @mk1r

mds1 commented:
>  Is there a section about setting up hardhat network in metamask somewhere? 

@mk1r There is in `app/README.md` which we link to in the contributing file

> and how do other contributors deal with not having VITE_FLEEK_STORAGE_API_KEY

Good question, @apbendi any idea if there's a free Fleek API key people can register for that would work just as well?

carletex commented on CONTRIBUTING.md:
```diff
@@ -155,6 +164,85 @@ git push origin HEAD # This will close the PR
+
+### Code Style Guide
+
+In general, the app follows the recommendations of the official [Vue Style Guide](https://vuejs.org/v2/style-guide/), which should be followed as much as possible.
```
Style Guide V3 link: https://v3.vuejs.org/style-guide/

mds1 replied:
Ah nice, done in 5a5d69d961814643f76f99a56755449d62564325
